auditlog_security/models/auditlog_rule.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2021 Therp B.V.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import exceptions, models, fields, api, modules, _
+from odoo.addons.auditlog.models.rule import FIELDS_BLACKLIST
+
+
+class AuditlogRule(models.Model):
+    _inherit = "auditlog.rule"
+
+    auditlog_line_access_rule_ids = fields.One2many(
+        "auditlog.line.access.rule", "auditlog_rule_id", ondelete="cascade"
+    )
+
+    @api.onchange("model_id")
+    def onchange_model_id(self):
+        # if model changes we must wipe out all field ids
+        self.auditlog_line_access_rule_ids.unlink()
+
+    @api.multi
+    def unlink(self):
+        lines = self.mapped("auditlog_line_access_rule_ids")
+        res = super(AuditlogRule, self).unlink()
+        if res:
+            lines.unlink()
+        return res
+
+    @api.multi
+    def subscribe(self):
+        super(AuditlogRule, self).subscribe()
+        act_window_model = self.env["ir.actions.act_window"]
+        for rule in self:
+            domain = (
+                "[('log_id.model_id', '=', %s), ('log_id.res_id', '=', "
+                "active_id),('field_id', 'in', %s)]"
```

Comment:
Using this domain requires access to the log records, which we want to prevent. So instead let's create a server action with Python and not a window action, and creating it will look something like:

```
'''
rule_id = {}   # filled into the server action string upon creation
fields_ids = auditlog_line_access_rule_ids.mapped('field_ids').ids
logs = env['auditlog.log'].sudo().search([('model_id', '=', %s), ('res_id', 'in', active_ids)])
domain = [('log_id', 'in', logs.ids), ('field_id', 'in', field_ids)]
action = {
                "name": _("View log lines"),
                "res_model": "auditlog.log.line",
                "src_model": rule.model_id.model,
                "binding_model_id": rule.model_id.id,
                "domain": domain
}
'''.format(rule.id)
```